### PR TITLE
Add SnapshotWithName

### PR DESCRIPTION
**Pull request by aeneasr**
SnapshotWithName is similar to SnapshotMulti without appending the function name.
It is useful when you need full control of the snapshot filename.

### CODE REVIEW

File: examples/advanced_test.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ func TestRawBytes(t *testing.T) {
 	result := bytes.NewBufferString("Hello advanced world!")
 	err := cupaloy.Snapshot(result.Bytes(), result, result.String())
 	if err != nil {
-		t.Fatal("New version of snapshot format should write out certain types directly", err)
+		t.Fatal("New version of snapshot format should wNite out certain types directly", err)
```
Comment (aaranmcguire) [inline examples/advanced_test.go:29]: I'm guessing this was done by mistake.

Comment (aeneasr) [reply examples/advanced_test.go:29]:
```suggestion
		t.Fatal("New version of snapshot format should write out certain types directly", err)
```

**Review by aaranmcguire — APPROVED** [review]
I'm not an owner or contributor of this repo, but I was about to write the same method so want to approve this :)

**Comment by aaranmcguire** [comment]
@bradleyjkemp Any chance this could be merged and released?

File: examples/advanced_test.go [hunk 2]
```diff
@@ -47,6 +47,14 @@ func TestConfig(t *testing.T) {
 	snapshotter.WithOptions(cupaloy.SnapshotSubdirectory("testdata")).SnapshotT(t, "Hello world!")
 }
 
+// SnapshotMulti allows a user to choose the full snapshot name
+func TestSnapshotMulti(t *testing.T) {
```
Comment (bradleyjkemp) [inline examples/advanced_test.go:51]:
```suggestion
func TestSnapshotWithName(t *testing.T) {
```

**Review by bradleyjkemp — COMMENTED** [review]
Just one tiny test naming change but otherwise this looks great!

Debated whether this should be an extra method or an option but I think this makes sense. Thanks for adding ❤️

**Review by bradleyjkemp — APPROVED** [review]

**Pull request merged**